### PR TITLE
Remove unused methods

### LINKED ISSUE #151: Spurious doctest error in rotation_matrix_error
In some setups we get a
```
File "flatsurf/geometry/matrix_2x2.py", line 208, in flatsurf.geometry.matrix_2x2.rotation_matrix_angle
Failed example:
    for _ in range(100):
        r = QQ.random_element(x=0,y=500)
        r -= r.floor()
        m = rot_matrix(r.numerator(), r.denominator())
        assert rotation_matrix_angle(m) == r
Exception raised:
    Traceback (most recent call last):
      File "/usr/share/miniconda/envs/test/lib/python3.7/site-packages/sage/doctest/forker.py", line 694, in _run
        self.compile_and_execute(example, compiler, test.globs)
      File "/usr/share/miniconda/envs/test/lib/python3.7/site-packages/sage/doctest/forker.py", line 1088, in compile_and_execute
        exec(compiled, globs)
      File "<doctest flatsurf.geometry.matrix_2x2.rotation_matrix_angle[5]>", line 5, in <module>
        assert rotation_matrix_angle(m) == r
      File "/home/runner/work/sage-flatsurf/sage-flatsurf/flatsurf/geometry/matrix_2x2.py", line 224, in rotation_matrix_angle
        raise RuntimeError
    RuntimeError
```

### PR DESCRIPTION
**Pull request by saraedum**
fixes #151.

### CODE REVIEW

File: flatsurf/geometry/matrix_2x2.py [hunk 1]
```diff
@@ -22,87 +22,14 @@
...
-    """
-    if not is_similarity(m):
-        raise ValueError("the matrix must be a similarity")
-
-    det = m.det()
-
-    if not det.is_square():
-        if not AA.has_coerce_map_from(m.base_ring()):
-            l = map(number_field_to_AA,m.list())
-            M = MatrixSpace(AA,2)
```
Comment (saraedum) [inline flatsurf/geometry/matrix_2x2.py:96]: Note that `MatrixSpace` is not defined. So this code path would not have worked.

File: flatsurf/geometry/matrix_2x2.py [hunk 2]
```diff
@@ -183,62 +110,6 @@ def similarity_from_vectors(u,v):
...
-
-        This is using floating point arithmetic and might be wrong.
-    """
-    e0,e1 = r.change_ring(CDF).eigenvalues()
-    m0 = (e0.log() / 2 / CDF.pi()).imag()
-    m1 = (e1.log() / 2 / CDF.pi()).imag()
-    r0 = RR(m0).nearby_rational(max_denominator=10000)
-    r1 = RR(m1).nearby_rational(max_denominator=10000)
-    if r0 != -r1:
-        raise RuntimeError
```
Comment (saraedum) [inline flatsurf/geometry/matrix_2x2.py:224]: This assertion sometimes failed, see #151.

**Comment by saraedum** [comment]
@wphooper @videlec any reason we should keep these?

**Review by videlec — APPROVED** [review]

File: flatsurf/geometry/matrix_2x2.py (hunk 2 above)
Comment (videlec) [reply flatsurf/geometry/matrix_2x2.py:224]: It currently fails on master!

**Pull request merged**